crates/sui-replay/src/replay.rs

```diff
@@ -779,8 +779,21 @@ impl LocalExec {
 
         trace!(target: "replay_gas_info", "{}", Pretty(&gas_status));
 
-        if let ProgrammableTransaction(pt) = transaction_kind {
-            trace!(target: "replay_ptb_info", "{}", Pretty(&pt));
+        if let ProgrammableTransaction(ref pt) = transaction_kind {
+            trace!(target: "replay_ptb_info", "{}", Pretty(&(pt.clone(), executor.dev_inspect_transaction(&self, protocol_config,
```

Comment:
It feels strange that we have to rerun the transaction for the devinspect results. 
Thoughts on just using dev inspect the first time?

I'm totally fine with running it twice if thats the conclusion, just figured it was worth discussing.

Reply:
I actually kinda like rerunning it here...

Reasoning being that I want the default to be as close to the "real" transaction execution flow as possible since if there's an issue that only arises in non-dev-inspect mode that will be harder to diagnose if the replay tool always uses dev-inspect for the replay.

Reply:
Yes, also thought about creating a 3rd execution mode for replay to avoid needing to run twice. But also I'm interested in keeping the transaction flow as similar as possible to the original, since that to me is one of the main value propositions of replay

Reply:
got to say that running it twice feels a bit odd to me